Without this patch, the TestFchmodat fails on our arm CI with:

	syscall_linux_test.go:139: Fchmodat: unexpected error: operation not permitted, expected EOPNOTSUPP

The "operation not permitted" means that EPERM was returned which
is likely due to the security policy of our CI container.

diff -upr go.orig/src/syscall/syscall_linux_test.go go/src/syscall/syscall_linux_test.go
--- go.orig/src/syscall/syscall_linux_test.go	2024-02-07 22:54:39.316022227 +0100
+++ go/src/syscall/syscall_linux_test.go	2024-02-07 22:56:05.104871102 +0100
@@ -135,7 +135,7 @@ func TestFchmodat(t *testing.T) {
 	}
 
 	err = syscall.Fchmodat(_AT_FDCWD, "symlink1", 0444, _AT_SYMLINK_NOFOLLOW)
-	if err != syscall.EOPNOTSUPP {
+	if !testenv.SyscallIsNotSupported(err) && err != syscall.EOPNOTSUPP {
 		t.Fatalf("Fchmodat: unexpected error: %v, expected EOPNOTSUPP", err)
 	}
 }
